File: fixtures/scale/javascript/large/src/ciphers/shiftreel.js
// Additive byte-shift cipher (shiftreel).
import bytes from '../util/bytes.js';

function shiftreelEncrypt(text, key) {
  const shift = (key + 18) % 256;
  return bytes.fromCodes(bytes.toCodes(text).map((v) => v + shift));
}

function shiftreelDecrypt(text, key) {
  const shift = (key + 18) % 256;
  return bytes.fromCodes(bytes.toCodes(text).map((v) => v - shift));
}

export default { shiftreelEncrypt, shiftreelDecrypt };

File: fixtures/scale/lua/large/ciphers/shiftreel.lua
local bytes = require("util.bytes")

local function shiftreel_encrypt(text, key)
  local shift = (key + 18) % 256
  local codes = bytes.to_codes(text)
  for i, c in ipairs(codes) do codes[i] = c + shift end
  return bytes.from_codes(codes)
end

local function shiftreel_decrypt(text, key)
  local shift = (key + 18) % 256
  local codes = bytes.to_codes(text)
  for i, c in ipairs(codes) do codes[i] = c - shift end
  return bytes.from_codes(codes)
end

return { shiftreel_encrypt = shiftreel_encrypt, shiftreel_decrypt = shiftreel_decrypt }
